### ISSUE TITLE
aws_service_discovery_service broken for HTTP namespaces

### ISSUE THREAD

**Issue by zilman** [terraform-providers/terraform-provider-aws#7016]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v0.11.10
AWS Provider 1.53.0

### Affected Resource(s)

* aws_service_discovery_http_namespace
* aws_service_discovery_service 

### Terraform Configuration Files

This will successfully create the http namespace:

```terraform
resource "aws_service_discovery_http_namespace" "test" {
  name        = "test"
  description = "test http namespace"
}
```

However, if you try to create a service:
```terraform

resource "aws_service_discovery_service" "example" {
  name = "example"

  dns_config {
    namespace_id = "${aws_service_discovery_http_namespace.test.id}"

    dns_records {
      ttl  = 10
      type = "A"
    }
  }
}
```

```terraform
* aws_service_discovery_service.example: InvalidInput: When you specify an HTTP service, you can't include a DnsConfig element.
```

Of course if you remove it you get:

```terraform
Error: aws_service_discovery_service.example: "dns_config": required field is not set
```
### Expected Behavior

Should be able to omit dns_config for services created under HTTP namespaces.

Reproducible with awscli:

```
aws servicediscovery create-service --name example --dns-config "NamespaceId=ns-xxxxxx,DnsRecords=[{Type=CNAME,TTL=10}],RoutingPolicy=WEIGHTED" 
```

```
An error occurred (InvalidInput) when calling the CreateService operation: When you specify an HTTP service, you can't include a DnsConfig element.
```

If you remove it:
```
aws: error: argument --dns-config is required
```

But not Boto3, because [this works](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/servicediscovery.html#ServiceDiscovery.Client.create_service):
```
client.create_service(Name='example', NamespaceId='ns-xxxxx')
```

This is because creating a Service and [registering an Instance](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/servicediscovery.html#ServiceDiscovery.Client.register_instance) are two distinct API calls. 

When registering instances for a service under an HTTP namespace there needs to be a way to pass a map of arbitrary attributes. I think it makes sense to add an instances resource.

### References

The [AWS ServiceDiscovery Docs ](https://docs.aws.amazon.com/cloud-map/latest/api/API_CreateService.html#API_CreateService_RequestSyntax) correctly state:

> DnsConfig
A complex type that contains information about the Amazon Route 53 records that you want AWS Cloud Map to create when you register an instance.
>
>Type: DnsConfig object
>
> **Required: No**

The [Terraform ServiceDiscovery Docs](https://www.terraform.io/docs/providers/aws/r/service_discovery_service.html) on the other hand:

> dns_config - (Required) A complex type that contains information about the resource record sets that you want Amazon Route 53 to create when you register an instance.

**Comment by bflad** [comment]
Support for API services (under HTTP namespaces) was added upstream during AWS re:Invent, but support has not been implemented in the existing Terraform resource yet. Marked as an enhancement.

**Comment by zilman** [comment]
@bflad is there anybody assigned to this or an eta? 
If not I might have time to contribute a PR next week.

**Comment by bflad** [comment]
Support for HTTP namespaces with the `aws_service_discovery_service` resource has been merged and will release with version 2.19.0 of the Terraform AWS Provider, likely later this week. Thanks, @saravanan30erd!

**Comment by bflad** [comment]
This has been released in [version 2.19.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2190-july-11-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!
